crates/node/p2p/src/gossip/handler.rs

```diff
@@ -68,10 +63,11 @@ impl Handler for BlockHandler {
         match decoded {
             Ok(envelope) => {
                 if self.block_valid(&envelope) {
-                    _ = self.block_sender.send(envelope);
```

Comment:
Big rookie mistake - we ignored the future here instead of attempting to send. Unfortunately this can fail, and unsafe blocks could be ignored.

Maybe what we want here is to hold the blocks in a cache instead and allow the gossip driver to pull any unsafe blocks periodically.

Alternatively, this handler is invoked directly from the gossip driver. Instead of having the block handler send the `OpNetworkPayloadEnvelope` through a channel, we can hoist this logic up. The block handler can actually just return the `OpNetworkPayloadEnvelope` itself, which could be bubbled up all the way to the network driver. Since the network driver runs the blocking loop, it can continuously attempt to send any unsafe blocks in its cache through the channel.

Reply:
Yeah throwing them into a queue makes sense for sure. We could have it push/pull, but ensuring that the unsafe payloads stick around until we can guarantee the engine actor has received them sounds ideal. The implementation detail I'm not super opinionated on - good callout 💯

Reply:
https://github.com/op-rs/kona/issues/1365